### PR TITLE
Disable UpdatingEditionProvider when running Enso CLI

### LINKED ISSUE #9231: Fail instead of trying to download anything
The Enso system and specification contains intricate description of various _downloading mechanism_ to get the right GraalVM, libraries, etc.

The current policy however is: **never download anything**.

Please modify the _downloading systems_ to fail immediately for now. Possibly hide the downloading functionality behind some option/environment variable.

### LINKED ISSUE #9753: enso-bundle tries to download missing engine
Daily [version from Apr 19, 2024](https://github.com/enso-org/enso/releases/download/2024.1.1-nightly.2024.4.19/)
demonstrates a problem. Download the right "enso-bundle-" first. If on Linux download [enso-bundle-2024.1.1-nightly.2024.4.19-linux-amd64.tar.gz](https://github.com/enso-org/enso/releases/download/2024.1.1-nightly.2024.4.19/enso-bundle-2024.1.1-nightly.2024.4.19-linux-amd64.tar.gz) and untar it. Then
```
$ git clone https://github.com/otulach/SCMrozdeleni
$ cd SCMrozdeleni
$ git checkout cc1f40174186b33a7b7ec43984bc49757f8be742
SCMrozdeleni$ ../../enso/bin/enso run src/Download.enso 
[WARN] [2024-04-19T15:35:30+02:00] [org.enso.launcher.cli.CLIRuntimeVersionManagementUserInterface] Engine 0.0.0-dev is missing.
Do you want to install the missing engine? [Y/n] 
```
**No**! I don't want to download anything. As written in #9231 - our current policy is to **never download anything**.

Expected behavior: don't ask to _install the missing engine_.

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Fixes #9231 by using `FileSystemEditionManager` and not `UpdatingEditionManager` to avoid any downloads of editions.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      style guides. 
- [x] Unit tests were updated to pass

### CODE REVIEW

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala [hunk 1]
```diff
@@ -158,7 +158,11 @@ class MainModule(serverConfig: LanguageServerConfig, logLevel: Level) {
   val distributionManager = new DistributionManager(environment)
 
   val editionProvider =
-    EditionManager.makeEditionProvider(distributionManager, Some(languageHome))
+    EditionManager.makeEditionProvider(
+      distributionManager,
+      Some(languageHome),
+      false
```
Comment (JaroslavTulach) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala:164]: Passing `false` as we want to disable updating in the production.

File: engine/language-server/src/test/scala/org/enso/languageserver/websocket/json/BaseServerTest.scala [hunk 2]
```diff
@@ -347,7 +347,8 @@ abstract class BaseServerTest
     val editionProvider =
       EditionManager.makeEditionProvider(
         distributionManager,
-        Some(languageHome)
+        Some(languageHome),
+        true
```
Comment (JaroslavTulach) [inline engine/language-server/src/test/scala/org/enso/languageserver/websocket/json/BaseServerTest.scala:351]: In tests, we can happily test the `UpdatingEditionProvider`.

File: engine/runner/src/main/scala/org/enso/runner/DependencyPreinstaller.scala [hunk 3]
```diff
@@ -43,7 +43,8 @@ object DependencyPreinstaller {
 
     val editionProvider = EditionManager.makeEditionProvider(
       distributionManager,
-      Some(languageHome)
+      Some(languageHome),
+      true
```
Comment (JaroslavTulach) [inline engine/runner/src/main/scala/org/enso/runner/DependencyPreinstaller.scala:47]: Preinstaller shall be able to download. Passing `true`.

File: lib/scala/edition-updater/src/main/scala/org/enso/editions/updater/EditionManager.scala [hunk 4]
```diff
@@ -77,6 +87,6 @@ object EditionManager {
     distributionManager: DistributionManager,
     languageHome: Option[LanguageHome] = None
   ): EditionManager = new EditionManager(
-    makeEditionProvider(distributionManager, languageHome)
+    makeEditionProvider(distributionManager, languageHome, false)
```
Comment (JaroslavTulach) [inline lib/scala/edition-updater/src/main/scala/org/enso/editions/updater/EditionManager.scala:90]: By default build non-updating edition manager in production.

**Review by radeusgd — APPROVED** [review]

**Review by Akirathan — APPROVED** [review]

**Review by 4e6 — APPROVED** [review]

**Review by hubertp — APPROVED** [review]

**Pull request merged**